R/check_inputs.R

```diff
@@ -0,0 +1,28 @@
+# Wrapper function to check inputs
+check_inputs <- function(weights, data, var_interest,
+                         at, cofint, base_rn, type, dof, vcov_mat){
+
+  # Warn if base_rn set but type != 'effects'
+  if(base_rn != 1 & type != 'effects')
+    warning("Setting base_rn when type == 'levels' is ignored.")
+
+  # Weights should be same length as data
+  if(!is.null(weights) & length(weights) != nrow(data))
+    stop('`weights` and `data` must be the same length.')
```

Comment:
data(margex)
margex$distance[c(1,3,5)] <- NA
margex$age[c(10, 24, 390)] <- NA
margex$y[c(12, 220)] <- NA
This + how `clean_glm_data` handles weights is a problem when you have missing values but don't want to specify weights, because `mod$prior.weights` is already subset to complete cases. E.g.
```
mm <- glm(y ~ as.factor(treatment) + age, data = margex, family = 'gaussian',
          weights = distance)
z1 <- modmarg::marg(mod = mm, var_interest = 'treatment', type = 'levels')[[1]]
Error in modmarg::marg(mod = mm, var_interest = "treatment", type = "levels") : 
  `weights` and `data` must be the same length.
```
What about something like this:

* `weights` defaults to `NULL`
* if `weights` is NULL, subset `data` on rownames using `names(mod$prior.weights)`
* if `weights` is not NULL, it has to be the length of `data` and subsetting happens as we have here?